theano/gof/cmodule.py

```diff
@@ -1893,7 +1893,7 @@ def join_options(init_part):
         # or 64 bit and compile accordingly. This step is ignored for ARM
         # architectures in order to make Theano compatible with the Raspberry
         # Pi, and Raspberry Pi 2.
-        if not any(['arm' in flag for flag in cxxflags]) and platform.machine() != 'armv7l':
+        if not any(['arm' in flag for flag in cxxflags]) and platform.machine() not in ['armv7l','armv6l']:
```

Comment:
Can you make a PR that will add 'aarch64' in the list at the end of this line? Just test it before, I think it will fix it.

Otherwise, instead of making a black list that don't support -m64, this could be changed to a white list that support it. There is much less arch that support it.